File: frontend/src/interfaces/async/request/piece/PieceUpdateRequestDto.ts
/*
 * (C) 2026. - Rafael Urben
 */

import type { PieceDto } from "@/api/generated/openapi/models/PieceDto.ts";

export default interface PieceUpdateRequestDto {
  title: PieceDto["title"];
  composer: PieceDto["composer"];
  year: PieceDto["year"];
  description: PieceDto["description"];
  difficulty: PieceDto["difficulty"];
  bpmRange: PieceDto["bpmRange"];
  isPublic: PieceDto["isPublic"];
}

File: backend/src/main/java/ch/rafaelurben/sheetmusiclearner/backend/io/async/dto/request/PieceUpdateRequestDto.java
/* (C) 2026 - Rafael Urben */
package ch.rafaelurben.sheetmusiclearner.backend.io.async.dto.request;

public record PieceUpdateRequestDto(
    String title,
    String composer,
    String year,
    String description,
    String difficulty,
    String bpmRange,
    Boolean isPublic) {}
